lib/active_admin/resource/attributes.rb

```diff
@@ -0,0 +1,44 @@
+module ActiveAdmin
+
+  class Resource
+    module Attributes
+
+      def default_attributes
+        resource_class.columns.each_with_object({}) do |c, attrs|
+          unless reject_col?(c)
```

Comment:
Actually, do we need this check at all? I applied the following patch and all features passed:

```diff
diff --git a/lib/active_admin/csv_builder.rb b/lib/active_admin/csv_builder.rb
index ba323302..36fcd8e8 100644
--- a/lib/active_admin/csv_builder.rb
+++ b/lib/active_admin/csv_builder.rb
@@ -21,7 +21,7 @@ module ActiveAdmin
     # resource's content columns
     def self.default_for_resource(resource)
       new resource: resource do
-        column :id
+        # column :id
         resource.content_columns.each { |c| column c }
       end
     end
diff --git a/lib/active_admin/resource/attributes.rb b/lib/active_admin/resource/attributes.rb
index 48063ddf..c587b72d 100644
--- a/lib/active_admin/resource/attributes.rb
+++ b/lib/active_admin/resource/attributes.rb
@@ -24,6 +24,7 @@ module ActiveAdmin
       end
 
       def reject_col?(c)
+        return false
         primary_col?(c) || sti_col?(c) || counter_cache_col?(c)
       end
 
@@ -41,4 +42,4 @@ module ActiveAdmin
 
     end
   end
-end
\ No newline at end of file
+end
diff --git a/lib/active_admin/views/index_as_table.rb b/lib/active_admin/views/index_as_table.rb
index 01323833..5da66c4f 100644
--- a/lib/active_admin/views/index_as_table.rb
+++ b/lib/active_admin/views/index_as_table.rb
@@ -232,7 +232,7 @@ module ActiveAdmin
       def default_table
         proc do
           selectable_column
-          id_column if resource_class.primary_key
+          # id_column if resource_class.primary_key
           active_admin_config.resource_columns.each do |attribute|
             column attribute
           end
```

I can see a point for skipping the inheritance column on STI models (we might want to test this), but I think the `counter_cache` columns can be useful by default, and that we don't really need to skip the primary key.